### PR TITLE
refactor: change `zmk-dev-generic-cache` into `zmk-dev-generic` image

### PR DESCRIPTION
**Pull request by innovaker**
Makes it easier to inspect the layers/image.

### CODE REVIEW

File: .github/workflows/containers.yml [hunk 1]
```diff
@@ -101,7 +101,7 @@ jobs:
         env:
           NS: ${{ env.docker-hub-namespace }}
           NSU: ${{ env.docker-hub-namespace-upstream }}
-          REPOSITORY: zmk-dev-generic-cache
+          REPOSITORY: zmk-dev-generic
```
Comment (petejohanson) [inline .github/workflows/containers.yml:104]: Is the repo rename actually relevant to the fix here?

Comment (innovaker) [reply .github/workflows/containers.yml:104]: yes, it's no longer just `cache`, it's now a fully fledged image such as `zmk-dev-arm` etc.

Comment (innovaker) [reply .github/workflows/containers.yml:104]: whilst the cache is now `inline` (like the other images), someone could use it as a base image (`FROM: zmk-dev-generic`).

Comment (petejohanson) [reply .github/workflows/containers.yml:104]: Thanks. Makes sense now.

**Review by petejohanson — APPROVED** [review]

**Pull request merged**